src/BoundaryConditions.h

```diff
@@ -56,7 +56,7 @@ inline const Func &func_like_to_func(const Func &func) {
 }
 
 template <typename T>
-inline NO_INLINE Func func_like_to_func(const T &func_like) {
+inline NO_INLINE Func func_like_to_func(T &func_like) {
```

Comment:
Should probably add a comment somewhere about why these can't be const refs -- people who haven't seen this specific PR will be confused in the future otherwise.